File: app/src/main/java/com/mnj/java_solutions/array/WaveArray28092023.java
package com.mnj.java_solutions.array;

public class WaveArray28092023 {

    public void convertToWave() {
        int[] arr = {1, 2, 3, 4, 5};
        int n = arr.length;

        for (int i = 1; i < n; i += 2) {
            int c = arr[i];
            arr[i] = arr[i - 1];
            arr[i - 1] = c;
        }
    }
}

File: app/src/main/java/com/mnj/java_solutions/array/WaveArray28092023.kt
package com.mnj.java_solutions.array

class WaveArray28092023 {
    fun convertToWave() {
        val arr = intArrayOf(1, 2, 3, 4, 5)
        val n = arr.size
        var i = 1
        while (i < n) {
            val c = arr[i]
            arr[i] = arr[i - 1]
            arr[i - 1] = c
            i += 2
        }
    }
}